app/utils/datepicker-validation.js

```diff
@@ -0,0 +1,43 @@
+export const EMPTY_DATE = 'EMPTY_DATE';
+export const INVALID_DATE = 'INVALID_DATE';
+export const MIN_DATE = 'MIN_DATE';
+export const MAX_DATE = 'MAX_DATE';
+
+export function validate(validation, allowEmpty = true, errorMessages) {
+  if (!validation.valid) {
+    switch (validation.error) {
+      case INVALID_DATE: {
+        return {
+          valid: false,
+          errorMessage: errorMessages?.invalidDate || 'Ongeldige datum',
+        };
+      }
+      case MAX_DATE: {
+        return {
+          valid: false,
+          errorMessage:
+            errorMessages?.maxDate ||
+            'Kies een datum tussen 01-01-1900 en vandaag',
+        };
+      }
+      case MIN_DATE: {
+        return {
+          valid: false,
+          errorMessage:
+            errorMessages?.minDate ||
+            'Kies een datum tussen 01-01-1900 en vandaag',
+        };
+      }
+      case EMPTY_DATE: {
+        return {
+          valid: allowEmpty,
+          errorMessages: !allowEmpty
+            ? errorMessages?.emptyDate || 'datum kan niet leeg zijn'
+            : null,
+        };
+      }
+    }
+  } else {
+    return { valid: true };
+  }
+}
```

Comment:
I didn't wanted to be dependent on ember-changeset-validations as it would produce more boilerplate everywhere for something that should be simple imo (validate the format of a date in a custom component)
this way the validations are optional, if you put an invalid date e.g 99-99-9999 and didn't specify you want to handle the error, the date will simply be emptied.